### ISSUE TITLE
📦 Update Kea version to the current supported stable release 2.0.2

### ISSUE THREAD

**Issue by mitchdawson1982** [ministryofjustice/staff-device-dhcp-server#183]
### User Story

As a member of the Cloudops team.
I need to ensure that the software versions of the Kea dhcp software are kept up to date.
So that we can ensure that our products remain within windows where the vendor will accept bug reports or questions on the given version. 

### Value / Purpose

The current version of the Kea DHCP software 1.8.2 goes end of life in July 2022. We therefor need to upgrade to the current stable version 2.0.2 which has a longer end of life in order to retain the ability to raise bugs against the software.

### Useful Contacts

Mitch Dawson, Touhidal Islam

### Additional Information

https://www.isc.org/kea/

2.0.2 Release Notes - https://downloads.isc.org/isc/kea/2.0.2/Kea-2.0.2-ReleaseNotes.txt

### Definition of Done

- [ ] Kea dhcp server software updated to 2.0.2
- [ ] Documentation has been written / updated
- [ ] README has been updated
- [ ] User docs have been updated
- [ ] Another team member has reviewed
- [ ] Tests are green

**Comment by mTouhid** [comment]
Currently upgrading to v2.0.2 has resulted in successful local testing. We see the following messages in the container logs which indicate the service has started successfully, notably HA is also working.
```
HA_CONFIGURATION_SUCCESSFUL HA hook library has been successfully configured
HA_INIT_OK loading High Availability hooks library successful
DHCP4_MULTI_THREADING_INFO enabled: yes, number of threads: 12, queue size: 792
HA_STATE_TRANSITION server transitions from WAITING to SYNCING state, partner state is WAITING
HA_LEASE_UPDATES_DISABLED lease updates will not be sent to the partner while in SYNCING state
HA_SYNC_START starting lease database synchronization with standby
HA_LEASES_SYNC_LEASE_PAGE_RECEIVED received 0 leases from standby
HA_SYNC_SUCCESSFUL lease database synchronization with standby completed successfully in 8.563 ms
HA_STATE_TRANSITION server transitions from SYNCING to READY state, partner state is WAITING
HA_LEASE_UPDATES_DISABLED lease updates will not be sent to the partner while in READY state
```

**However, we noticed this error when we interacted with the admin portal locally**
```
ERROR HA_RESUME_CLIENT_LISTENER_FAILED Resuming multi-threaded HTTP processing failed: HttpClient::resume - no thread pool
```

We do not observe that same behaviour when running version 1.8.2 locally.

We have blocked the pipeline to pre-prod and in development the service starts but HA is not working. We can see the following logs which confirm this: 

```
ERROR HA_RESUME_CLIENT_LISTENER_FAILED Resuming multi-threaded HTTP processing failed: HttpClient::resume - no thread pool
ERROR HA_PAUSE_CLIENT_LISTENER_FAILED Pausing multi-threaded HTTP processing failed: HttpClient::pause - no thread pool
```

**Comment by mitchdawson1982** [comment]
We have been able to successfully deploy Kea 2.0.2. After consulting the documentation it was noted that the HA service had received a big architectural change and required additional configuration parameters by way of of its own multithreading object in the config.json file.

Example:

"Dhcp4": {

    ...
    "hooks-libraries": [
        {
            "library": "/usr/lib/kea/hooks/libdhcp_lease_cmds.so",
            "parameters": { }
        },
        {
            "library": "/usr/lib/kea/hooks/libdhcp_ha.so",
            "parameters": {
                "high-availability": [ {
                    "this-server-name": "server1",
                    ...
                    "multi-threading": {
                        "enable-multi-threading": true,
                        "http-dedicated-listener": true,
                        "http-listener-threads": 4,
                        "http-client-threads": 4
                    },
                    ...
                    "peers": [
                      // This is the configuration of this server instance.
                      {
                          "name": "server1",
                          // This specifies the URL of our server instance. Since the
                          // HA+MT uses direct connection, the DHCPv4 server open its own
                          // socket. Note it must be different than the one used by the
                          // CA (typically 8000). In this example, 8001 is used.
                          "url": "http://192.0.2.1:8001/",
                          // This server is primary. The other one must be secondary.
                          "role": "primary"
                      },
                      // This is the configuration of our HA peer.
                      {
                          "name": "server2",
                          // This specifies the URL of our server instance. Since the
                          // HA+MT uses direct connection, the DHCPv4 server open its own
                          // socket. Note it must be different than the one used by the
                          // CA (typically 8000). In this example, 8001 is used.
                          "url": "http://192.0.2.2:8001/",
                          // The partner is a secondary. Our is primary.
                          "role": "secondary"
                      }
                    ...

As documented here - https://kea.readthedocs.io/en/kea-2.0.0/arm/hooks.html#multi-threaded-configuration-ha-mt

**Comment by Gary-H9** [comment]
Opened DHCP Pipeline to Production. 
(upgrading mysql -> v12 and updated kea to v2.0.2)

Opened Admin pipeline to Production.
(This published code which published a new config.json with added new HA elements mentioned above)

Saved an entry in admin to reload config.

Confirmed the config was updated by manually downloading and examining it.

We also confirmed there were no HA messages or any other errors in the logs.
